Automattic/wp-calypso
# Jetpack Connect: Block blacklisted sites

#### Changes proposed in this Pull Request

* Introduce a blacklisted site notice and its corresponding notice type.
* Refine `checkUrl` action error to contain only what we might ever need from it.
* Store the JPC authorization errors in the state.
* Introduce a `isSiteBlacklistedError` selector.
* Block blacklisted sites in URL entry step. 
* Block blacklisted sites in auth step.

#### Preview

Blocking blacklisted sites in the site URL entry step:
![](https://cldup.com/6rXT6_bgrb.png)

Blocking blacklisted sites in the authorization step:
![](https://cldup.com/Xl6aCP-DZb.png)

#### Testing instructions

* Checkout this branch.
* If you haven't done that yet, follow instructions in the following diff: 
* Create a basic JN site from [this link](http://jurassic.ninja/create).
* Connect the site, selecting a free plan.
* Blacklist the site by adding the "Black" flag in the RC of the blog.
* Disconnect the site.
* Attempt to connect the site, starting from wp-admin (add `&calypso_env=development` to the connect button URL).
* Verify that in the auth step you can see the error as shown on the screenshot.
* Go to http://calypso.localhost:3000/jetpack/connect
* Input the site URL and submit the form.
* Verify you can see the error in the same page, before the auth process has started.
* Test both the Jetpack and the Calypso connection flows with a non-blacklisted site, verify they still work as expected.

This fixes 3155-gh-jpop-issues.
